pkg/restore/restore.go

```diff
@@ -1249,12 +1249,31 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs
 	}
-	if isAlreadyExistsError {
-		fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
-		if err != nil {
-			ctx.log.Infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
-			warnings.Add(namespace, err)
-			return warnings, errs
+
+	// check if we want to treat the error as a warning, in some cases the creation call might not get executed due to object API validations
+	// and Velero might not get the already exists error type but in reality the object already exists
+	objectExists := false
+	var fromCluster *unstructured.Unstructured
+
+	if restoreErr != nil && !isAlreadyExistsError {
+		// check for the existence of the object in cluster, if no error then it implies that object exists
+		// and if err then we want to fallthrough and do another get call later
+		fromCluster, err = resourceClient.Get(name, metav1.GetOptions{})
+		if err == nil {
+			objectExists = true
+		}
+	}
+
+	if isAlreadyExistsError || objectExists {
+		// do a get call if we did not run this previously i.e.
+		// we've only run this for errors other than isAlreadyExistError
+		if fromCluster == nil {
+			fromCluster, err = resourceClient.Get(name, metav1.GetOptions{})
+			if err != nil {
+				ctx.log.Infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
```

Comment:
I know this is copied from existing code, but seems it's more consistent (e.g with the code in line 1247 ) if we merge this `err` into `errors`.
We may also wanna change the log level.

Reply:
@reasonerjt Hmm. I'm not sure, but yeah, you may be right. So this would be a change in behavior for situations not otherwise modified by this PR. So the scenario we're talking about is that the `Create` failed due to the object already existing, but now when we do a `Get` call (to determine whether the in-cluster object is different from the from-backup object), the Get call fails. Ordinarily this shouldn't happen. Off the top of my head, possible causes are:
1) Something outside of Velero just created this resource at almost the same time as Velero tried to restore and we're in a race scenario where a second Create fails due to an already existing object but the API server isn't yet finding it via Get.
2) A bug or malfunction in the APIServer

We're currently flagging this as a restore warning. If we want to stick with that, then we should probably change the log level from Info to Warn.

The other possibility is we bump this up to a restore error and the log level to Error as well.

@shubham-pampattiwar what do you think?

Reply:
@reasonerjt @sseago I agree with changing the log level to error and adding this error to errors (as the next piece of workflow depends on the object fetched from the cluster)

Reply:
Done, updated the PR.